Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInput/CreditCardInputField.swift

```diff
@@ -0,0 +1,92 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+
+/// For the Credit Card feature, this enum holds four cases that describe different parts of a credit card.
+enum CreditCardInputType {
+    case name, number, expiration, securityCode
+}
+
+struct CreditCardInputField: View {
+    struct Colors {
+        let errorColor: Color
+        let titleColor: Color
+        let textFieldColor: Color
+    }
+
+    @Binding var text: String
+    @ObservedObject var viewModel: CreditCardInputViewModel
+    let creditCardInputType: CreditCardInputType
+    var colors: Colors
+
+    init(creditCardInputType: CreditCardInputType,
+         text: Binding<String>,
+         colors: Colors) {
+        self.creditCardInputType = creditCardInputType
+        self.viewModel = CreditCardInputViewModel(fieldType: creditCardInputType)
+        self._text = text
+        self.colors = colors
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            getInputFieldViewWith(inputType: creditCardInputType)
+        }.padding(.leading, 16)
+    }
+
+    @ViewBuilder private func getInputFieldViewWith(inputType: CreditCardInputType) -> some View {
+        Text(viewModel.fieldHeadline)
+            .font(.subheadline)
+            .foregroundColor(colors.titleColor)
+        TextField("", text: $text)
+            .font(.body)
+            .foregroundColor(colors.textFieldColor)
+            .keyboardType(viewModel.keyboardType)
+            .onChange(of: text) { [oldValue = text] newValue in
+                switch inputType {
```

Comment:
for readability I think this logic should be a separate function probably in the viewModel so it can be easily tested